site/docs/v1/tech/core-concepts/search.adoc

```diff
@@ -84,3 +84,32 @@ Optionally, you can also reindex via API.
 
 - link:/docs/v1/tech/apis/system#rebuild-the-elasticsearch-index[Rebuild Index API]
 - link:/docs/v1/tech/apis/system#retrieve-the-status-of-an-index-rebuild[Index Status API]
+
+== Pagination
+
+Whichever search engine you are using, you will likely need to paginate the results when running a query that matches many users or entities.
+
+Use the `numberOfResults` and `startRow` parameters to do so.
+
+Here's pseudocode to get all results of a query:
+
+----
+startRow = 0
+numberOfResults = 25
+fullresults = [] // new array
+results = client.search(query, startRow, numberOfResults)
+count = results.length
+fullresults.append(results)
+
+while (count > 0) {
+  startRow = startRow + count 
+  results = client.search(query, startRow, numberOfResults)
+  fullresults.append(results)
+  count = results.length
+}
+
+// process fullresults
+----
+
+You may also set `numberOfResults` to a higher number (500 or 5000, for example) to retrieve more results.
+However, processing results 25 or 50 at a time has less impact on the FusionAuth system.
```

Comment:
> However, processing a smaller set of results (25 or 50 at a time) is the preferred mechanism as this is more performant.

maybe?

Reply:
I changed to 'However, processing results 25 or 50 at a time has less impact on the FusionAuth system.' which is more accurate in all cases, I believe.